### PR TITLE
ciao-storage: Improve error logging from rbd commands

### PR DESCRIPTION
**Pull request by rbradford**
Report the output from rbd commands in the log if the command errors
out (non-zero exit code.)

Fixes: #642

Signed-off-by: Rob Bradford <robert.bradford@intel.com>

### CODE REVIEW

**Comment by markdryan** [comment]
@tpepper This is generally the case but not universally true.  The exec functions like Output and CombinedOutput can return non-nil slices for their first return value even if their second return value is a non-nil error.  There are other examples in the standard library such as io.Reader.Read.

**Comment by rbradford** [comment]
@tpepper I did check this by forcing some failure paths .e.g. by trying to delete the CNCI image which rbd wouldn't let you do because it's actively in use.

In those cases I got an !nil err as well as output.

File: ciao-storage/ceph.go [hunk 1]
```diff
@@ -166,16 +170,22 @@ func (d CephDriver) MapVolumeToNode(volumeUUID string) (string, error) {
...
+	}
+	return nil
 }
 
 // GetVolumeMapping returns a map of volumeUUID to mapped devices.
 func (d CephDriver) GetVolumeMapping() (map[string][]string, error) {
 	args := append(d.getCredentials(), "showmapped", "--format", "json")
 	cmd := exec.Command("rbd", args...)
-	data, err := cmd.Output()
+	data, err := cmd.CombinedOutput()
```
Comment (markdryan) [inline ciao-storage/ceph.go]: This is changing the behaviour of the function.  The old version was reading the first line of Stdout.  The new code now reads the first line of combined Stdout and Stderr which might not be the same thing.  Probably best to continue to use Output, but to retrieve the error information by

err.(*ExitError).Stderr

Comment (rbradford) [inline ciao-storage/ceph.go]: Ah @markdryan  that's a nice solution. I'd have hoped that rbd did the right right of not writing random stuff to stderr with a zero exit code. But this fix is the safest.

File: ciao-storage/ceph.go [hunk 2]
```diff
@@ -175,7 +185,7 @@ func (d CephDriver) GetVolumeMapping() (map[string][]string, error) {
 	cmd := exec.Command("rbd", args...)
 	data, err := cmd.Output()
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("Error when running: %v: %v: %s", cmd.Args, err, err.(*exec.ExitError).Stderr)
```
Comment (markdryan) [inline ciao-storage/ceph.go]: @rbradford Sorry this is my fault, but reading the docs again the error is not guaranteed to be of type ExitError.  If it's not the cast will panic.

> Any returned error will usually be of type *ExitError. If c.Stderr was nil, Output populates ExitError.Stderr. 

So we should probably write
```
var output []byte
if err, ok := err.(*exec.ExitError); ok {
  output = err.Stderr
}
```

to be safe.

**Pull request merged**